protocol/contracts/templegold/SpiceAuction.sol

```diff
@@ -0,0 +1,501 @@
+ * Bid and auction tokens could change per auction. These are set in `AuctionConfig`. 
+ * Config is set before the next auction starts using `setAuctionConfig` by DAO execution.
+ */
+contract SpiceAuction is ISpiceAuction, AuctionBase, ReentrancyGuard {
+    using SafeERC20 for IERC20;
+    using TempleMath for uint256;
+    using EpochLib for IAuctionBase.EpochInfo;
+    using OptionsBuilder for bytes;
+
+    /// @notice Spice auction contracts are set up for 2 tokens. Either token can be bid or sell token for a given auction
+    address public immutable override spiceToken;
+    /// @notice Temple GOLD
+    address public immutable override templeGold;
+    /// @notice DAO contract to execute configurations update
+    address public override daoExecutor;
+    /// @notice operator
+    address public override operator;
+
+    /// @notice Auctions run for minimum 1 week
+    uint32 public constant MINIMUM_AUCTION_PERIOD = 1 weeks;
+    /// @notice Maximum wait period between last and next auctions
+    uint32 public constant MAXIMUM_AUCTION_WAIT_PERIOD = 90 days;
+    /// @notice Maximum auction duration
+    uint32 public constant MAXIMUM_AUCTION_DURATION = 30 days;
+    /// @notice Arbitrum One layer zero EID
+    uint32 private immutable _arbitrumOneLzEid;
+    /// @notice Arbitrum One chain ID
+    uint32 private immutable _mintChainId;
+    /// @notice Max gas limit for use by executor in calling `lzReceive`
+    uint32 public override lzReceiveExecutorGas;
+
+    /// @notice Name of this Spice Bazaar auction
+    string public override name;
+
+    /// @notice Last time auction was started. For zero auctions, it is the contract deploy timestamp
+    uint256 private immutable _deployTimestamp;
+
+    /// @notice Keep track of the different configurations for each auction
+    mapping(uint256 auctionId => SpiceAuctionConfig config) public auctionConfigs;
+    /// @notice Keep track of claimed amounts per auction token. This helps calculate amount of auction tokens for next epoch.
+    mapping(address token => uint256 amount) private _claimedAuctionTokens;
+    /// @notice Keep track of total allocation per auction token
+    mapping(address token => uint256 amount) private _totalAuctionTokenAllocation;
+    /// @notice Keep track of redeemed and notified epochs
+    mapping(uint256 epochId => bool redeemed) public override redeemedEpochs;
+
+    constructor(
+        address _templeGold,
+        address _spiceToken,
+        address _daoExecutor,
+        address _operator,
+        uint32 _arbEid,
+        uint32 mintChainId_,
+        string memory _name
+    ) {
+        spiceToken = _spiceToken;
+        daoExecutor = _daoExecutor;
+        operator = _operator;
+        templeGold = _templeGold;
+        _arbitrumOneLzEid = _arbEid;
+        _mintChainId = mintChainId_;
+        name = _name;
+        _deployTimestamp = block.timestamp;
+        lzReceiveExecutorGas = 85_412;
```

Comment:
is this number up to date with the latest implementation changes?

Reply:
updated

Reply:
why does this number has such a high precission? would it make sense to make it a bit higher just to have some buffer?